server/src/main/java/org/opensearch/search/fetch/subphase/highlight/UnifiedHighlighter.java

```diff
@@ -133,13 +135,40 @@ public HighlightField highlight(FieldHighlightContext fieldContext) throws IOExc
         return new HighlightField(fieldContext.fieldName, Text.convertFromStringArray(fragments));
     }
 
+    public AnalyzerWrapper getLimitedOffsetAnalyzer(Analyzer a, int limit) {
```

Comment:
You could probably simplify it a bit:

```
    public AnalyzerWrapper getLimitedOffsetAnalyzer(Analyzer a, int limit) {
        return new AnalyzerWrapper(a.getReuseStrategy()) {
            @Override
            protected Analyzer getWrappedAnalyzer(String fieldName) {
                return a;
            }

            @Override
            protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComponents components) {
                return new TokenStreamComponents(
                    components.getSource(),
                    new LimitTokenOffsetFilter(components.getTokenStream(), limit)
                );
            }
        };
    }

```

Reply:
Do you have any suggestion, this is only used to limit the offset of the tokens generated by the analyzer, ignoring all the text after that.

Reply:
I mean, you don't need to introduce any fields, just use arguments:

```
 private Analyzer old = a;
            private int maxOffset = limit;
```

could be removed.

Reply:
For sure,you are correct.